libmariadb/CMakeLists.txt

```diff
@@ -309,7 +309,9 @@ ELSE()
   IF(ICONV_INCLUDE_DIR)
      INCLUDE_DIRECTORIES(BEFORE ${ICONV_INCLUDE_DIR}) 
   ENDIF()
-  ADD_DEFINITIONS(-DLIBICONV_PLUG)
+  IF(NOT CMAKE_SYSTEM_NAME MATCHES AIX)
+    ADD_DEFINITIONS(-DLIBICONV_PLUG)
+  ENDIF()
```

Comment:
iconv related, but not this change:

I got link errors using the 3.1 branch as updated in libmariadb under mariadb-server
```
Scanning dependencies of target libmariadb
[ 12%] Linking C shared library libmariadb
ld: 0711-317 ERROR: Undefined symbol: .libiconv_open
ld: 0711-317 ERROR: Undefined symbol: .libiconv
ld: 0711-317 ERROR: Undefined symbol: .libiconv_close
ld: 0711-345 Use the -bloadmap or -bnoquiet option to obtain more information.
```

It turns out  `/usr/lib/libiconv.a` doesn't define these symbols,  `/usr/linux/lib/libiconv.a`  does.

I used the ugly hack of
```
--- a/cmake/FindIconv.cmake
+++ b/cmake/FindIconv.cmake
@@ -28,6 +28,9 @@ ELSEIF(APPLE)
                /usr/lib/
                NO_CMAKE_SYSTEM_PATH)
     SET(ICONV_EXTERNAL TRUE)
+ELSEIF(CMAKE_SYSTEM_NAME MATCHES "AIX")
+  SET(ICONV_LIBRARIES -liconv)
+  SET(ICONV_EXTERNAL TRUE)
 ELSE()
   find_library(ICONV_LIBRARIES NAMES iconv libiconv libiconv-2)
   IF(ICONV_LIBRARIES)
```
To work around this. Do you have any ideas on making this more robust?

Reply:
We usually use libraries built by opensource teams (in /usr/linux/lib or /opt/freeware/lib), even if they are available on base system (except ssl and crypto). So, I have not faced this issue. Your hack seems to be compatible with both version of libiconv. I do not have idea to make it more robust.

Reply:
Thanks. Highlighting it has made a push to remove it as a dependency anyway.